projects/Mallard/src/services/logging.ts

```diff
@@ -154,19 +154,25 @@ class Logging extends AsyncQueue {
         return response
     }
 
-    async postLog(log: object[]) {
+    async postLogs() {
         try {
-            await this.postLogToService(log)
-            this.numberOfAttempts = 0
+            const { isConnected } = await NetInfo.fetch()
+            // Only attempt if we are connected, otherwise wait till next time
+            if (isConnected) {
+                const logsToPost = await this.getQueuedItems()
+                await this.postLogToService(logsToPost)
```

Comment:
maybe worth adding a check here or in `postLogToService` to check that there's at least one element in logsToPost